uqcsbot/scripts/weather.py

```diff
@@ -0,0 +1,214 @@
+from uqcsbot import bot, Command
+from urllib.request import urlopen
+import xml.etree.ElementTree as ET
+from datetime import datetime as DT
+
+
+def get_xml(state: str):
+    """
+    get BOM data as an XML for a given state
+    """
+    source = {"NSW": "IDN11060", "ACT": "IDN11060", "NT": "IDD10207", "QLD": "IDQ11295",
+              "SA": "IDS10044", "TAS": "IDT16710", "VIC": "IDV10753", "WA": "IDW14199"}
+    try:
+        data = urlopen("ftp://ftp.bom.gov.au/anon/gen/fwo/{}.xml".format(source[state]))
+        root = ET.fromstring(data.read())
+    except Exception:
+        return None
+    return root
+
+
+def process_arguments(arguments: str):
+    """
+    process the arguments given to !weather, dividing them into state, location and future
+    uses default of QLD, Brisbane and 0 if not given
+    """
+    args = arguments.split(" ") if arguments else []
+    if args and args[-1].lstrip('-+').isnumeric():
+        future = int(args.pop())
+    else:
+        future = 0
+
+    # get location
+    if args:
+        if args[0].upper() in ["NSW", "ACT", "NT", "QLD", "SA", "TAS", "VIC", "WA"]:
+            state = args.pop(0).upper()
+        else:
+            state = "QLD"
+        location = " ".join(args)
+    else:
+        state = "QLD"
+        location = "Brisbane"
+
+    return state, location, future
+
+
+def find_location(root: ET.Element, location: str, future: int):
+    """
+    , returns the XML for a given the location and how far into the future
+    """
+    node = root.find(".//area[@description='{}']".format(location))
+    if node is None:
+        return None, "Location Not Found"
+    if node.get("type") != "location":
+        return None, "Location Given Is Region"
+    node = node.find(".//forecast-period[@index='{}']".format(future))
+    if node is None:
+        return None, "No Forecast Available For That Day"
+    return node, None
+
+
+def response_header(node: ET.Element, location: str):
+    """
+    returns the response header, in the form "{Location}'s Weather Forcast For {Day}"
+    """
+    forcast_date = DT.strptime("".join(node.get('start-time-local')
+                                       .rsplit(":", 1)), "%Y-%m-%dT%H:%M:%S%z").date()
+    today_date = DT.now().date()
+    date_delta = (forcast_date - today_date).days
+    if date_delta == 0:
+        date_name = "Today"
+    elif date_delta == 1:
+        date_name = "Tomorrow"
+    elif date_delta == -1:
+        # can happen during the witching hours
+        date_name = "Yesterday"
+    else:
+        date_name = forcast_date.strftime("%A")
+    return "*{}'s Weather Forcast For {}*".format(date_name, location)
```

Comment:
`f"*{date_name}'s Weather Forecast For {location}*"`

Reply:
Fixed.